packages/@aws-cdk/aws-sagemaker/lib/endpoint.ts

```diff
@@ -0,0 +1,448 @@
+    return this.metric('AWS/SageMaker', responseCode, props);
+  }
+
+  public metricDiskUtilization(props?: cloudwatch.MetricOptions): cloudwatch.Metric {
+    return this.metric('/aws/sagemaker/Endpoints', 'DiskUtilization', {
+      statistic: 'Average',
+      ...props,
+    });
+  }
+
+  public metricCpuUtilization(props?: cloudwatch.MetricOptions): cloudwatch.Metric {
+    return this.metric('/aws/sagemaker/Endpoints', 'CPUUtilization', {
+      statistic: 'Average',
+      ...props,
+    });
+  }
+
+  public metricMemoryUtilization(props?: cloudwatch.MetricOptions): cloudwatch.Metric {
+    return this.metric('/aws/sagemaker/Endpoints', 'MemoryUtilization', {
+      statistic: 'Average',
+      ...props,
+    });
+  }
+
+  public metricGpuUtilization(props?: cloudwatch.MetricOptions): cloudwatch.Metric {
+    return this.metric('/aws/sagemaker/Endpoints', 'GPUUtilization', {
+      statistic: 'Average',
+      ...props,
+    });
+  }
+
+  public metricGpuMemoryUtilization(props?: cloudwatch.MetricOptions): cloudwatch.Metric {
+    return this.metric('/aws/sagemaker/Endpoints', 'GPUMemoryUtilization', {
+      statistic: 'Average',
+      ...props,
+    });
+  }
+
+  public autoScaleInstanceCount(scalingProps: appscaling.EnableScalingProps): ScalableInstanceCount {
+    const errors: string[] = [];
+    if (scalingProps.minCapacity && scalingProps.minCapacity > this.initialInstanceCount) {
+      errors.push(`minCapacity cannot be greater than initial instance count: ${this.initialInstanceCount}`);
+    }
+    if (scalingProps.maxCapacity && scalingProps.maxCapacity < this.initialInstanceCount) {
+      errors.push(`maxCapacity cannot be less than initial instance count: ${this.initialInstanceCount}`);
+    }
+    if (BURSTABLE_INSTANCE_TYPE_PREFIXES.some(prefix => this.instanceType.toString().startsWith(prefix))) {
+      errors.push(`AutoScaling not supported for burstable instance types like ${this.instanceType}`);
+    }
+    if (this.scalableInstanceCount) {
+      errors.push('AutoScaling of task count already enabled for this service');
+    }
+
+    if (errors.length > 0) {
+      throw new Error(`Invalid Application Auto Scaling configuration: ${errors.join(EOL)}`);
+    }
+
+    return this.scalableInstanceCount = new ScalableInstanceCount(this.endpoint, 'InstanceCount', {
+      serviceNamespace: appscaling.ServiceNamespace.SAGEMAKER,
+      resourceId: `endpoint/${this.endpoint.endpointName}/variant/${this.variantName}`,
+      dimension: 'sagemaker:variant:DesiredInstanceCount',
+      role: this.makeScalingRole(),
+      minCapacity: scalingProps.minCapacity || this.initialInstanceCount,
+      maxCapacity: scalingProps.maxCapacity || this.initialInstanceCount,
+    });
+  }
+
+  /**
+   * Return the service linked role that will be used for AutoScaling
+   *
+   * Documentation is available here: https://docs.aws.amazon.com/autoscaling/application/userguide/application-auto-scaling-service-linked-roles.html
```

Comment:
The documentation here is good, and helped me find what I was looking for. But let's pull this into the docstring anyway. Application autoscaling creates service-linked roles for you, that is why they are being hardcoded here (maybe that's obvious for app autoscaling folks, but as a generic CDK guy i was unaware). 

"Return the pre-created service linked role..." or something